src/receive_imf.rs

```diff
@@ -1473,15 +1473,29 @@ async fn add_parts(
     for part in &mime_parser.parts {
         if part.is_reaction {
             let reaction_str = simplify::remove_footers(part.msg.as_str());
+            let reaction = Reaction::from(reaction_str.as_str());
             set_msg_reaction(
                 context,
                 mime_in_reply_to,
                 orig_chat_id.unwrap_or_default(),
                 from_id,
                 sort_timestamp,
-                Reaction::from(reaction_str.as_str()),
+                reaction.clone(),
             )
             .await?;
+            let is_fresh_reaction = !seen && !fetching_existing_messages;
+            if mime_parser.incoming && is_fresh_reaction && !reaction.is_empty() {
+                if let Some((msg_id, _)) = rfc724_mid_exists(context, mime_in_reply_to).await? {
```

Comment:
I suggest to move this code to `set_msg_reaction()` somehow, at least there's already a call to `rfc724_mid_exists()`, though still the `msgs` table isn't accessed there, so `Message::load_from_db()` may be needed, but i think the check whether the message is outgoing can be merged into some existing SQL statement there

Reply:
hm, not sure adding complexity to the function and if it is worth optimising here too much, at the level of SQL.

but `set_msg_reaction()` could return `MsgId`, so that the additional call to `rfc724_mid_exists()` is not needed

Reply:
i initially thought, `set_msg_reaction()` was used also elsewhere, but it is only called from `receive_imf.rs` - so, yes, your advice makes lots of sense, saves one db call and keeps `receive_imf.rs` clean :) thanks a lot!